File: Passenger.cpp
// Passenger.cpp
#include "Passenger.hpp"

namespace InfSystBusStation {

    Passenger::Passenger(System::String^ fullName, System::String^ gender,
        System::String^ passportSeries, System::String^ passportNumber,
        System::String^ email, System::String^ phone)
        : People(fullName, gender, passportSeries, passportNumber, email),
        phoneNumber(phone) {
    }

    void Passenger::PrintInfo() {
        People::PrintInfo(); // Вызов базового метода
        System::Console::WriteLine("Телефон: {0}", phoneNumber);
    }

    System::String^ Passenger::GetFullInfo() {
        return "Пассажир: " + GetFullName() +
            " | Паспорт: " + GetPassportSeries() + " " + GetPassportNumber() +
            " | Телефон: " + phoneNumber;
    }

    double Passenger::CalculateDiscount() {
        return 0.0; // Базовая скидка
    }
}

File: Passenger.java
public class Passenger extends People {
    private String passportSeries;
    private String passportNumber;
    private String email;

    public Passenger(String surname, String name, String middleName,
                    String passportSeries, String passportNumber, String email) {
        super(surname, name, middleName, passportSeries, passportNumber, email);
        this.passportSeries = passportSeries;
        this.passportNumber = passportNumber;
        this.email = email;
    }

    // Геттеры для данных пассажира
    public String getPassportSeries() {
        return passportSeries;
    }

    public String getPassportNumber() {
        return passportNumber;
    }

    public String getEmail() {
        return email;
    }

    public void printPassengerInfo() {
        System.out.println("=== Passenger Information ===");
        System.out.println("Full name: " + getFullName());
        System.out.println("Passport: " + passportSeries + " " + passportNumber);
        System.out.println("Email: " + email);
        System.out.println("=============================");
    }
}